shared/modules/selectors/smart-transactions.ts

```diff
@@ -89,13 +94,21 @@ const getIsAllowedRpcUrlForSmartTransactions = (
   return rpcUrl?.hostname?.endsWith('.infura.io');
 };
 
+const hasNonZeroBalance = (state: SmartTransactionsMetaMaskState) => {
+  const selectedAccount = getSelectedAccount(
+    state as unknown as MultichainState,
+  );
+  return new BigNumber(selectedAccount?.balance || '0x0', 16).gt(0);
```

Comment:
Nit: We could use `BigInt` here instead of `BigNumber`, and avoid the import

```suggestion
  returnBigInt(selectedAccount?.balance || '0x0') > 0;
```